connectors/kibana.py

```diff
@@ -348,10 +352,13 @@ async def upsert_index(es, index, docs=None, mappings=None, settings=None):
     if docs is None:
         return
     # TODO: bulk
-    doc_id = 1
-    for doc in docs:
-        await es.client.index(index=index, id=doc_id, document=doc)
-        doc_id += 1
+
+    if doc_ids is None:
+        for doc_id, doc in enumerate(docs, start=1):
+            await es.client.index(index=index, id=doc_id, document=doc)
+    else:
+        for doc, doc_id in zip(docs, doc_ids, strict=True):
+            await es.client.index(index=index, id=doc_id, document=doc)
```

Comment:
I remember I've seen this change already - is it because of merge by chance?

Reply:
Nope, this was all me 😅 
Because I switched off Native, that meant that the config.yml needed to specify connector IDs. Which meant that the IDs couldn't be non-deterministic. So I had to modify this logic to pass along the pre-configured connector ID.